lib/csu/common-cheri/crt_init_globals.c

```diff
@@ -145,12 +145,13 @@ crt_init_globals(const Elf_Phdr *phdr, long phnum,
 #else
 		data_cap = cheri_getdefault();
 #endif
-		data_cap = cheri_clearperm(data_cap, CHERI_PERM_EXECUTE);
+		data_cap = cheri_clearperm(data_cap,
+		   CHERI_PERM_EXECUTE | CHERI_PERM_SW_VMEM);
 
 		code_cap = cheri_getpcc();
```

Comment:
I wonder, does the kernel give this SW_VMEM currently?...

Reply:
It does not, testing this to make sure but I think it is not required. If so I'll remove this.

Reply:
Uhm, looks like without it riscv fails the test, although the kernel should be stripping VMEM from what I've seen..

Reply:
In `imgact_elf.c` the AT_PHDR capability is given out with SW_VMEM permission. Perhaps we want to change this but I'm not sure whether it is required in some case.

Reply:
I seems interesting that it has been explicitly set for candperm, but I agree. I'll add a commit to strip it and remove this change in crt.

Reply:
I think something is using the SW_VMEM permission from PHDR to `mprotect` relro, so probably this is still required.

Reply:
Presumably the something is rtld?  You can't do that on relro until after applying the relocations which is done in userland, so that makes sense I guess.  We should perhaps document why we have to leave SW_VMEM enabled here via a comment though?

Reply:
Yeah that something is rtld, I meant something within rtld, which I did not fully track down. I think leaving it with a comment makes sense.